channels/tasks.py

```diff
@@ -131,3 +138,73 @@ def populate_subscriptions_and_roles(self):
         ]
     )
     raise self.replace(results)
+
+
+@app.task
+def populate_post_post_type(ids):
+    """
+    Populates Post.post_type for a range of posts
+
+    Args:
+        ids (list of int): list of post ids
+    """
+    client = Api(User.objects.get(username=settings.INDEXING_API_USERNAME))
+
+    for post in Post.objects.filter(id__in=ids).iterator():
+        if post.post_type:
+            continue
+
+        submission = client.get_submission(post.post_id)
+        if submission.url:
+            post.post_type = LINK_TYPE_LINK
+        elif post.article:
+            post.post_type = EXTENDED_POST_TYPE_ARTICLE
+        else:
+            post.post_type = LINK_TYPE_SELF
+
+        post.savE()
+
+
+@app.task
+def populate_channel_post_type(channel_ids):
+    """
+    Populates Post.post_type for a range of posts
+
+    Args:
+        channel_ids (list of int): list of post ids
+    """
+    client = Api(User.objects.get(username=settings.INDEXING_API_USERNAME))
+
+    for channel in Channel.objects.filter(id__in=channel_ids).iterator():
+        if channel.allowed_post_types:
+            continue
+
+        subreddit = client.get_subreddit(channel.display_name)
```

Comment:
Use `channel.name` instead?  When this task runs via the `backpopulate_post_tasks` command, it throws an 
`AttributeError: 'Channel' object has no attribute 'display_name'`